src/main/java/it/gov/pagopa/onboarding/workflow/service/OnboardingServiceImpl.java

```diff
@@ -248,26 +272,32 @@ private void checkDates(InitiativeDTO initiativeDTO, Onboarding onboarding) {
     }
   }
 
-  private void checkBudget(InitiativeDTO initiativeDTO, Onboarding onboarding) {
-    InitiativeGeneralDTO generalInfo = initiativeDTO.getGeneral();
-    BigDecimal totalBudget = generalInfo.getBudget();
-    BigDecimal beneficiaryBudget = generalInfo.getBeneficiaryBudget();
-    int onboardedCitizen = onboardingRepository.countByInitiativeIdAndStatus(initiativeDTO.getInitiativeId(),
-            OnboardingWorkflowConstants.ONBOARDING_OK) + 1;
+  private void checkBudget(String initiativeId, Onboarding onboarding) {
+    try {
+      InitiativeStatusDTO initiativeStatusDTO = admissibilityRestConnector.getInitiativeStatus(
+          initiativeId);
+      if (initiativeStatusDTO.isBudgetAvailable() && initiativeStatusDTO.getStatus()
+          .equals(OnboardingWorkflowConstants.PUBLISHED)) {
+        return;
+      }
+        LocalDateTime localDateTime = LocalDateTime.now();
+        onboarding.setStatus(OnboardingWorkflowConstants.ONBOARDING_KO);
+        onboarding.setOnboardingKODate(localDateTime);
+        onboarding.setUpdateDate(localDateTime);
+        onboarding.setDetailKO(OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED);
+        onboardingRepository.save(onboarding);
+        auditUtilities.logOnboardingKOWithReason(onboarding.getInitiativeId(),
+            onboarding.getUserId(), onboarding.getChannel(),
+            OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED_MSG);
+        throw new OnboardingWorkflowException(HttpStatus.FORBIDDEN.value(),
+            OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED_MSG,
+            OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED);
+
 
-    BigDecimal budgetUsed = beneficiaryBudget.multiply(BigDecimal.valueOf(onboardedCitizen));
-    if (budgetUsed.compareTo(totalBudget) > 0){
-      LocalDateTime localDateTime = LocalDateTime.now();
-      onboarding.setStatus(OnboardingWorkflowConstants.ONBOARDING_KO);
-      onboarding.setOnboardingKODate(localDateTime);
-      onboarding.setUpdateDate(localDateTime);
-      onboarding.setDetailKO(OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED);
-      onboardingRepository.save(onboarding);
-      auditUtilities.logOnboardingKOWithReason(onboarding.getInitiativeId(), onboarding.getUserId(), onboarding.getChannel(),
-              OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED_MSG);
-      throw new OnboardingWorkflowException(HttpStatus.FORBIDDEN.value(),
-              OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED_MSG,
-              OnboardingWorkflowConstants.ERROR_BUDGET_TERMINATED);
+    } catch (FeignException e) {
```

Comment:
Let's manage the exception inside the connector, so that we have a cleaner service :)